src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Shadow/PlatformShadowEffect.android.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Xamarin.Forms.Platform.Android;
+using Xamarin.Forms;
+using Android.Views;
+using AView = Android.Views.View;
+using Android.Content;
+using Android.OS;
+using System.ComponentModel;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.CommunityToolkit.Android.Effects;
+using Android.Widget;
+
+[assembly: ExportEffect(typeof(PlatformShadowEffect), nameof(ShadowEffect))]
+
+namespace Xamarin.CommunityToolkit.Android.Effects
+{
+	public class PlatformShadowEffect : PlatformEffect
+	{
+		const float defaultRadius = 10f;
+
+		const float defaultOpacity = 1f;
+
+		AView View => Control ?? Container;
+
+		protected override void OnAttached()
+		{
+			if (View == null)
+				return;
+
+			Update();
+		}
+
+		protected override void OnDetached()
+		{
+			if (View == null)
+				return;
+
+			View.SetElevation(0);
+		}
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			if (View == null)
+				return;
+
+			switch (args.PropertyName)
+			{
+				case nameof(ShadowEffect.ColorPropertyName):
+				case nameof(ShadowEffect.OpacityPropertyName):
+				case nameof(ShadowEffect.RadiusPropertyName):
+				case nameof(ShadowEffect.OffsetXPropertyName):
+				case nameof(ShadowEffect.OffsetYPropertyName):
+					View.Invalidate();
+					Update();
+					break;
+			}
+		}
+
+		void Update()
+		{
+			if (Build.VERSION.SdkInt < BuildVersionCodes.Lollipop)
+				return;
+
+			var color = ShadowEffect.GetColor(Element);
+			if (color.IsDefault)
+			{
+				View.SetElevation(0);
+				return;
+			}
+
+			var radius = (float)ShadowEffect.GetRadius(Element);
+			if (radius < 0)
+				radius = defaultRadius;
+
+			var opacity = ShadowEffect.GetOpacity(Element);
+			if (opacity < 0)
+				opacity = defaultOpacity;
+
+			var androidColor = color.MultiplyAlpha(opacity).ToAndroid();
+
+			if (View is TextView textView)
+			{
+				var offsetX = (float)ShadowEffect.GetOffsetX(Element);
+				var offsetY = (float)ShadowEffect.GetOffsetY(Element);
+				textView.SetShadowLayer(radius, offsetX, offsetY, androidColor);
```

Comment:
`SetShadowLayer` is only supported on text when hardware acceleration is on. Hardware acceleration is on by default when targetSdk=14 or higher. Can we add an API Level check here?. An easy workaround is to put the View in a software layer: `LAYER_TYPE_SOFTWARE`.

Reply:
Hmm, yes, we can. But does it make sense? As I know Forms support API 19 and above. (and recommend to use API 21 and above)